src/Symfony/Component/Mailer/Bridge/MicrosoftGraph/Transport/MicrosoftGraphTransportFactory.php

```diff
@@ -0,0 +1,65 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Mailer\Bridge\MicrosoftGraph\Transport;
+
+use Symfony\Component\Mailer\Bridge\MicrosoftGraph\TokenManager;
+use Symfony\Component\Mailer\Exception\IncompleteDsnException;
+use Symfony\Component\Mailer\Exception\InvalidArgumentException;
+use Symfony\Component\Mailer\Exception\UnsupportedSchemeException;
+use Symfony\Component\Mailer\Transport\AbstractTransportFactory;
+use Symfony\Component\Mailer\Transport\Dsn;
+use Symfony\Component\Mailer\Transport\TransportInterface;
+
+class MicrosoftGraphTransportFactory extends AbstractTransportFactory
+{
+    public function create(Dsn $dsn): TransportInterface
+    {
+        if ('microsoft+graphapi' !== $dsn->getScheme()) {
+            throw new UnsupportedSchemeException($dsn, 'microsoft graph api', $this->getSupportedSchemes());
+        }
+
+        $tenantId = $dsn->getOption('tenantId');
+        if (null === $tenantId) {
+            throw new IncompleteDsnException("Transport 'microsoft+graphapi' requires the 'tenant' option.");
+        }
+
+        $graphEndpoint = $dsn->getHost();
+        $authEndpoint = $dsn->getOption('authEndpoint');
+        if ('default' === $graphEndpoint) {
+            $graphEndpoint = 'graph.microsoft.com';
+            if (null === $authEndpoint) {
+                $authEndpoint = 'login.microsoftonline.com';
+            }
+        }
+
+        if (null === $authEndpoint) {
+            throw new IncompleteDsnException("Transport 'microsoft+graphapi' requires the 'authEndpoint' option when not using the default graph endpoint.");
+        }
+
+        if (0 !== preg_match('#^https?://#', $authEndpoint)) {
+            throw new InvalidArgumentException('Auth endpoint needs to be provided without http(s)://.');
+        }
+
+        if (0 !== preg_match('#^https?://#', $graphEndpoint)) {
```

Comment:
```suggestion
        if (!preg_match('#^https?://#', $graphEndpoint)) {
```

Reply:
I've applied this to be `if (preg_match('#^https?://#', $graphEndpoint)) {` (so, without the `!`), but that does mean that the `false` isn't handled anymore.

> preg_match() returns 1 if the pattern matches given subject, 0 if it does not, or [false](https://www.php.net/manual/en/reserved.constants.php#constant.false) on failure.